templates/zerver/accounts_home.html

```diff
@@ -78,7 +78,8 @@ <h1 class="get-started">{{ _("Sign up for Zulip") }}</h1>
                             <form class="form-inline" action="{{ backend.signup_url }}" method="get">
                                 <input type='hidden' name='multiuse_object_key' value='{{ multiuse_object_key }}' />
                                 <button id="register_{{ backend.button_id_suffix }}" class="login-social-button full-width"
-                                  {% if backend.display_icon %} style="background-image:url({{ backend.display_icon }})"  {% endif %}>
+                                  {% if backend.display_name == "Apple" %} style="background-image: url(https://appleid.cdn-apple.com/appleid/button/?width=328&height=46&color=white&locale={{ apple_locale }}&type=continue);"
```

Comment:
Why can't this computation be done in the called and used to populate `backend.display_icon`?

Reply:
We need to use different button url on login and signup page, plus the `{{ apple_locale }}` thing needs to be here and it's generated based on the request (based on language of the browser. Doesn't seem to be any good way to have this code anywhere in the backend class, and of all other places to put this Apple-specific code this is probably the least awful one. In an ideal world Apple would restrain themselves from forcing absurd rules on people at least when it comes to buttons, and we would be to use our regular buttons with a simple Apple logo...